### PR TITLE
fixed splash screen slightly jumps before hiding

### LINKED ISSUE #2542: [hold for payment 7 May] Loading screen logo slightly shifts up before fading away.
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
Loading screen logo is stable. 

## Actual Result:
Loading screen logo slightly shifts up before fading away.

## Action Performed:
1. Kill the chat app if open 
2. Relaunch the app
## Workaround:
No need for workaround. Visual issue. 

## Platform:
Where is this issue occurring?

 Web
 iOS
 Android ✔️ 
 Desktop App
 Mobile Web

**Version Number:** 1.0.29-0

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/44479856/115789564-55352b80-a393-11eb-92f7-1f7e60325dbe.mp4

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___ 
From @quinthar https://expensify.slack.com/archives/C01GTK53T8Q/p1619062191006900

>ISSUE: the new loading screen starts with the icon in the middle, but then the shifts ever so slightly up right before disappearing. It reliably does this every time I "cold open" the app on Android

### PR DESCRIPTION
**Pull request by parasharrajat**
Please review @Beamanator.

### Details
Previously, We manage to fix this issue by making the status bar translucent from the android theme https://github.com/Expensify/Expensify.cash/pull/2567. but it caused a regression https://github.com/Expensify/Expensify.cash/issues/2577#issuecomment-827244046.

#### New Solution
1. Another way to fix this is by setting the status bar translucent, only after the Splash screen is hidden.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes #2542

### Tests / QA Steps
1. Kill the chat app if open
2. Relaunch the app
3. Notice that the Splash screen should fade away without jumping up or down.

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [x] Android (affected platform)

#### Android

https://user-images.githubusercontent.com/24370807/116290968-aee18f80-a7b1-11eb-88c7-c6fdcc07d205.mp4

### CODE REVIEW

File: src/Expensify.js [hunk 1]
```diff
@@ -124,6 +124,14 @@ class Expensify extends PureComponent {
         return lodashGet(this.props, 'session.authToken', null);
     }
 
+    hideSplash() {
+        BootSplash.hide({fade: true}).then(() => {
+            // To prevent the splash from shifting positions we set status bar translucent after splash is hidden.
+            // on IOS it has no effect.
+            StatusBar.setTranslucent(true);
```
Comment (Beamanator) [inline src/Expensify.js:131]: `src/components/CustomStatusBar/index.android.js` mentions `StatusBar.setTranslucent` is only supported in android - is that going to cause any problems here?

File: src/components/CustomStatusBar/index.android.js [hunk 2]
```diff
@@ -9,7 +9,6 @@ export default class CustomStatusBar extends React.Component {
     componentDidMount() {
         StatusBar.setBarStyle('dark-content', true);
         StatusBar.setBackgroundColor('transparent', true);
-        StatusBar.setTranslucent(true);
```
Comment (Beamanator) [inline src/components/CustomStatusBar/index.android.js:12]: If we remove `setTranslucent` here, should we also remove it from the comment above that mentions it?

**Comment by Beamanator** [comment]
@marcaaron would you mind also checking out these changes related to BootSplash since you added that code recently?

File: src/Expensify.js (hunk 1 above)
Comment (parasharrajat) [reply src/Expensify.js:131]: no its will only show a warning in development build on IOS. in production it simply does return;
https://github.com/facebook/react-native/blob/d6cd2e6559ff8698833dc277810e2e7e80af760a/Libraries/Components/StatusBar/StatusBar.js#L336

File: src/components/CustomStatusBar/index.android.js (hunk 2 above)
Comment (parasharrajat) [reply src/components/CustomStatusBar/index.android.js:12]: Oh yeah. Didn't notice it.

**Review by Dal-Papa — DISMISSED** [review]

**Comment by parasharrajat** [comment]
@Beamanator Updated. Thanks.

**Review by marcaaron — APPROVED** [review]

**Pull request merged**